### PR TITLE
BUG: Fix read_excel w/parse_cols & empty dataset

### LINKED ISSUE #9208: read_excel drops columns when reading a one row file with a MultiIndex
I'm seeing the weirdest error.

When I read an .xlsx excel file with only one row (example below)...
image
... and try to set the `index_col`, then pandas incorrectly reads the columns:

I.e. for a file with only one row, column headers A, B, C, D, E, F

``` python
pandas.read_excel(filename, 'Sheet1', index_col=['A', 'B', 'C'])
```

Yields
image
(I get an index with columns A, B, C, and a dataframe with columns B, D, F - what happened to E?)

Note that if I don't use `index_col`, I get a dataframe with columns A-F as expected. Likewise, if I add a row of actual data, the dataframe loads as expected.

Relevant versions:
`pandas version 0.15.1 running on 2.7.7 |Anaconda 2.0.1 (64-bit)| (default, Jun 11 2014, 10:40:02) [MSC v.1500 64 bit (AMD64)]`
`xlrd.__VERSION__`=0.9.3

### PR DESCRIPTION
**Pull request by gfyoung**
Closes #9208.

### CODE REVIEW

File: pandas/io/excel.py [hunk 1]
```diff
@@ -634,14 +634,17 @@ def _parse_cell(cell_contents, cell_typ):
...
-                    last = data[offset][col]
-
-                    for row in range(offset + 1, len(data)):
-                        if data[row][col] == '' or data[row][col] is None:
-                            data[row][col] = last
-                        else:
-                            last = data[row][col]
+                # Check if we have an empty dataset
+                # before trying to collect data.
+                if offset < len(data):
```
Comment (TomAugspurger) [inline pandas/io/excel.py]: Do we have a test where `offset` is equal to `len(data)`? Just want to make sure we're not off by one, and it's not clear to be that we're not (without having read the code).

**Review by TomAugspurger — APPROVED** [review]
LGTM: just one small question.

What are the changes to the excel file?

File: pandas/io/excel.py (hunk 1 above)
Comment (gfyoung) [reply pandas/io/excel.py]: > What are the changes to the excel file?

To add a test that addresses your question:

> Do we have a test where offset is equal to len(data)?

Don't worry, I accounted for the off-by-one 😉

**Comment by TomAugspurger** [comment]
That's what I was hoping :) +1

**Pull request merged**